hazelcast/src/main/java/com/hazelcast/core/RingbufferStore.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.core;
+
+import java.util.Map;
+
+/**
+ * Ringbuffer store makes a ring buffer backed by a central data store; such as database, disk, etc.
+ *
+ * @param <T> ring buffer item type
+ */
+public interface RingbufferStore<T> {
+
+    /**
+     * Stores one item with it's corresponding sequence.
+     *
+     * @param sequence the sequence ID of the data
+     * @param data     the value of the data to store
+     */
+    void store(long sequence, T data);
+
+    /**
+     * Stores multiple entries. Implementation of this method can optimize the store operation.
+     *
+     * @param map map of entries to store
+     */
+    void storeAll(Map<Long, T> map);
```

Comment:
Is there a reason there needs to be a map? Are the items not a sequence of items? So you would only need an array an an initial sequence for the first item.